tests/kernel/common fails to build on xtensa / ESP32

commit 43553da9b2bb9e537185695899fb5184c3b17ebe

Toolchain: ZEPHYR_GCC_VARIANT=espressif

```
...
[ 63%] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/pinmux/pinmux_esp32.c.obj
[ 64%] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/timer/sys_clock_init.c.obj
zephyr.git/tests/kernel/common/src/intmath.c:50:30: error: enumerator value for '__build_assert37' is not an integer constant
zephyr.git/tests/kernel/common/src/intmath.c:52:30: error: enumerator value for '__build_assert41' is not an integer constant
 ROLLOVER_CHECK(int, 2147483647, -2147483648);
                              ^
```

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -18,11 +18,11 @@
 #define BUILD_ASSERT(EXPR) static_assert(EXPR, "")
 #define BUILD_ASSERT_MSG(EXPR, MSG) static_assert(EXPR, MSG)
 /*
- * GCC 4.6 and higher have _Static_assert built in, and its output is
- * easier to understand than the common BUILD_ASSERT macros.
+ * GCC 4.6 and higher have the C11 _Static_assert built in, and its
+ * output is easier to understand than the common BUILD_ASSERT macros.
  */
-#elif (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && \
-	(__STDC_VERSION__) > 199901L
+#elif (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) || \
+	(__STDC_VERSION__) >= 201100
 #define BUILD_ASSERT(EXPR) _Static_assert(EXPR, "")
 #define BUILD_ASSERT_MSG(EXPR, MSG) _Static_assert(EXPR, MSG)
 #endif

